core/src/main/java/com/orientechnologies/orient/core/command/script/OScriptManager.java

```diff
@@ -87,10 +88,20 @@
   public OScriptManager() {
     scriptEngineManager = new ScriptEngineManager();
 
+    final boolean useGraal = OGlobalConfiguration.SCRIPT_POLYGLOT_USE_GRAAL.getValueAsBoolean();
+
     executorsFactories.put(
-        "javascript", (lang) -> new OJavascriptScriptExecutor(lang, new OScriptTransformerImpl()));
+        "javascript",
+        (lang) ->
+            useGraal
+                ? new OPolyglotScriptExecutor(lang, new OScriptTransformerImpl())
+                : new OJsr223ScriptExecutor(lang, new OScriptTransformerImpl()));
```

Comment:
will the the Jsr223 be "loadable" in future java releases ? we may need to automatically bail out or even provide a pluggable module for the different jvms, so far this can be good as first step, but we may need to move the jsr223 in a external module if create problems with recent jvms.

Reply:
It's not really clear if and when jsr223 will be completely out from the jdk. I think it's ok to keep both with a switch and when a new JDK comes out without that package we could remove it and support only the new graalvm.

Reply:
I think creating 2 external modules for this is overkill.